back/engines/commercial/idea_assignment/app/services/idea_assignment/patches/side_fx_idea_service.rb

```diff
@@ -3,12 +3,14 @@ module Patches
     module SideFxIdeaService
       def before_update(idea, user)
         super
-        idea.assignee = nil if idea.project_id_changed? && !::ProjectPolicy.new(idea.assignee, idea.project).moderate?
+        if idea.project_id_changed? && idea.assignee && !UserRoleService.new.can_moderate_project?(idea.project, idea.assignee)
+          idea.assignee = nil
+        end
       end
 
       def after_update(idea, user)
         super
-        return unless idea.assignee_id_previously_changed?
+        return if !idea.assignee_id_previously_changed?
```

Comment:
Is this a Rubocop auto correct? I find `unless` quite readable.

Reply:
I personally find unless less readable 😅  I find the sentence "return, unless the assignee changed" harder to understand than "return if the assignee hasn't changed". "Unless" for me sounds like you assume the event is not going to happen, which I find less intuitive when you're writing/reading the line of code. In this case though the assignee is not likely to change, so I would be okay with using "unless" here, but stop conditions are often used for exceptional cases. I also like to have the negation very explicit with "if !".